### DISCUSSION TITLE
Resource preloaded but not used warning

### DISCUSSION THREAD

**Discussion by N1ckSL** [remix-run/remix#4038] [Q&A]
### After a short period of time (clicking some elements, accessing some internal links), I get this console warning:
The resource <URL> was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.

Sometimes I get 12 of them, sometimes up to 21. This is what it looks like:
![image](https://user-images.githubusercontent.com/54579507/185914506-fcc2db5f-9ce0-4018-87a8-24b21556bba0.png)

I've looked for an answer and got onto a similar issue, but it was about [Gatsby](https://github.com/gatsbyjs/gatsby/issues/14872)

Warnings suggest that an appropriate 'as' is required, but the thing is that I included it:
```
// /app/root.tsx
export const links: LinksFunction = () => {
  return [
    { rel: 'stylesheet', href: spinnersStyles, as: 'style' },
    { rel: 'stylesheet', href: styles, as: 'style' },
    { rel: 'stylesheet', href: untitledFont, as: 'font' },
  ];
};
```

After adding `crossOrigin: 'anonymous'`, which was suggested in  [Gatsby](https://github.com/gatsbyjs/gatsby/issues/14872) discussion I still get this warning (but only when I go to another page - for example `/login`):
![image](https://user-images.githubusercontent.com/54579507/185916892-9110fd49-bdae-490c-9df8-1f367ef24d60.png)

Could this warning occur due to the fact that all `stylesheets` are included in the `root` file, or there would be another reason?

**Comment by Ryongyon** [comment]
[![I6rOU.png](https://s1.328888.xyz/2022/09/22/I6rOU.png)](https://imgloc.com/i/I6rOU)
Remix example has the same problem

**Reply by Ryongyon** [reply]
My own project has the same problem, even though my Link component does not use prefetch. I don't know whether this is a Remix problem or a browser problem. Can anyone answer it

**Comment by sergiodxa** [comment]
This is a browser thing, if the time between the preload and the time the browser see the resource being used is not short enough it will show that warning, which could mean the resource may be being request again by the browser.

The only solution is to ensure your preloaded asset is used short enough, e.g. don't preload a file that the page doesn't need until JS load.

---

Also this:

```ts
export const links: LinksFunction = () => {
  return [
    { rel: 'stylesheet', href: spinnersStyles, as: 'style' },
    { rel: 'stylesheet', href: styles, as: 'style' },
    { rel: 'stylesheet', href: untitledFont, as: 'font' },
  ];
};
```

Is not correct, the `rel` needs to be `preload` not `stylesheet` for `as` to work, and a stylesheet can be a font at the same time.

**Reply by Ryongyon** [reply]
Why do I get this warning even if I don't use preload ?

**Reply by Ryongyon** [reply]
I think this is a bug, because I have defined a global style in the root.tsx , and I will receive this warning after I jump to the route。 I put forward an issue. Could you come and have a look? [https://github.com/remix-run/remix/issues/4257](https://github.com/remix-run/remix/issues/4257)
